lib/Doctrine/ORM/Query/SqlWalker.php

```diff
@@ -854,9 +856,40 @@ public function walkIdentificationVariableDeclaration($identificationVariableDec
      */
     public function walkIndexBy($indexBy)
     {
-        $pathExpression = $indexBy->simpleStateFieldPathExpression;
+        $pathExpression = $indexBy->singleValuedPathExpression;
         $alias          = $pathExpression->identificationVariable;
-        $field          = $pathExpression->field;
+
+        /** @var Query\AST\PathExpression $pathExpression */
```

Comment:
We prefer `assert()` over `/** @var */`

Reply:
Sorry, I wasn't entirely clear of the purpose of some of the docblocks like this. Perhaps the `@var` is redundant as the variable isn't a function parameter?

Reply:
If you don't need it for auto-complete or SA tools like PHPStan or Psalm, it should be removed, otherwise please use `assert($pathExpression instanceof PathExpression)` (and import that class of course). It will have the same effect but might be checked in some setups, which makes it better (less likely to rot than a comment).